src/components/Calculator.js

```diff
@@ -1,239 +1,236 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import '../index';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      operators: '+-x%÷',
-      numbers: '0123456789',
-      total: '',
-      next: '',
-      operation: '',
-      completed: false,
-      finalObject: {},
-      expression: '',
-    };
-  }
+const Calculator = () => {
+  const operators = '+-x%÷';
+  const numbers = '0123456789';
+  const [total, setTotal] = useState('');
+  const [next, setNext] = useState('');
+  const [operation, setOperation] = useState('');
+  const [expression, setExpression] = useState('0');
+  const [completed, setCompleted] = useState(false);
+  const [finalObject, setFinalObject] = useState({});
 
-  componentDidUpdate(previousProps, previousState) {
+  const passObject = (param) => {
+    setFinalObject(param);
+    setCompleted(true);
+  };
+
+  useEffect((previousProps, previousState) => {
     if (
-      previousState.total !== this.state.total
-      || previousState.next !== this.state.next
-      || previousState.operation !== this.state.operation
+      previousState !== total
+      || previousState !== next
+      || previousState !== operation
     ) {
       const object = {
-        total: this.state.total,
-        next: this.state.next,
-        operation: this.state.operation,
+        total,
+        next,
+        operation,
       };
 
       if (object.total && object.next && object.operation) {
-        this.passObject(object);
+        passObject(object);
       }
     }
-  }
+  });
 
-  passObject = (param) => {
-    this.setState({ finalObject: param });
-    this.setState({ completed: true });
-  };
-
-  handleButtonEvent = (event) => {
-    if (this.state.operators.includes(event.target.name)) {
-      this.setState({ operation: event.target.name });
-      this.setState({ expression: event.target.name });
+  const handleButtonEvent = (event) => {
+    if (operators.includes(event.target.name)) {
+      setOperation(event.target.name);
+      setExpression(event.target.name);
     }
 
-    if (this.state.numbers.includes(event.target.name)) {
-      if (this.state.total === '') {
-        this.setState({ total: event.target.name });
-        this.setState({ expression: event.target.name });
+    if (numbers.includes(event.target.name)) {
+      if (total === '') {
+        setTotal(event.target.name);
+        setExpression(event.target.name);
       } else {
-        this.setState({ next: event.target.name });
-        const exp = this.state.expression;
-        this.setState({
-          expression: `${exp} ${event.target.name}`,
-        });
+        setNext(event.target.name);
+        // const exp = expression;
```

Comment:
- [ ] [ OPTIONAL ] Please do not leave commented code in your codebase. It makes your work messy and unprofessional. Use git history if you need to review a different version of your code.